src/spec/create_extension_spec.py

```diff
@@ -1,47 +1,394 @@
+                                                                             neurodata_type_inc='LabNotebookNumericalValues'
+                                                                             ),
+                                                              NWBDatasetSpec(doc='Numerical labnotebook keys: First row is the name, second row is the unit and third row is the tolerance. '
+                                                                             'Columns are the running index. '
+                                                                             'See also https://alleninstitute.github.io/MIES/labnotebook-docs.html.',
+                                                                             neurodata_type_inc='LabNotebookNumericalKeys'
+                                                                             ),
+                                                              NWBDatasetSpec(doc='Textual labnotebook keys: First row is the name, second row is the unit and third row is the tolerance. '
+                                                                             'Columns are the running index. '
+                                                                             'See also https://alleninstitute.github.io/MIES/labnotebook-docs.html.',
+                                                                             neurodata_type_inc='LabNotebookTextualKeys'
+                                                                             ),
+                                                              NWBDatasetSpec(doc='Textual labnotebook values: Rows are the '
+                                                                             'running index, Columns hold the different entry names, Layers (up to nine) hold '
+                                                                             'the headstage dependent data in the first 8 and the headstage independent data in the 9th layer. '
+                                                                             'See also https://alleninstitute.github.io/MIES/labnotebook-docs.html.',
+                                                                             neurodata_type_inc='LabNotebookTextualValues'
+                                                                             )
+                                                              ]
+                                                    )
+                                       ]
+                               )
+
+    # Since 05cb3ffe (NWB: Export single test pulses as well on interactive use, 2017-11-07)
+
+    stored_testpulses = NWBDatasetSpec(doc='Raw AD testpulse data',
+                                       quantity='*',
+                                       neurodata_type_def='TestpulseRawData',
+                                       neurodata_type_inc='NWBData',
+                                       shape=(None))
```

Comment:
I believe you need a comma in the tuple if this is intended to be a 1D array of any length and not `shape=None` which means any shape.

Reply:
@rly Got it.